spec/policy/rate_limit/rate_limit_spec.lua

```diff
@@ -0,0 +1,175 @@
+      assert.spy(ngx_exit_spy).was_called_with(500)
+    end)
+    it('invalid limiter values', function()
+      local config = {
+        limiters = {
+          {name = "fixed_window", key = 'test1', count = 0, window = 10}
+        },
+        redis_url = 'redis://localhost:6379/1'
+      }
+      local rate_limit_policy = RateLimitPolicy.new(config)
+      rate_limit_policy:access()
+      assert.spy(ngx_exit_spy).was_called_with(500)
+    end)
+    it('no redis url', function()
+      local config = {
+        limiters = {
+          {name = "connections", key = 'test1', conn = 20, burst = 10, delay = 0.5},
+          {name = "leaky_bucket", key = 'test2', rate = 18, burst = 9},
+          {name = "fixed_window", key = 'test3', count = 10, window = 10}
+        }
+      }
+      local rate_limit_policy = RateLimitPolicy.new(config)
+      rate_limit_policy:access()
+    end)
+    it('invalid redis url', function()
+      local config = {
+        limiters = {
+          {name = "connections", key = 'test1', conn = 20, burst = 10, delay = 0.5}
+        },
+        redis_url = 'redis://invalidhost:6379/1'
+      }
+      local rate_limit_policy = RateLimitPolicy.new(config)
+      rate_limit_policy:access()
+      assert.spy(ngx_exit_spy).was_called_with(500)
+    end)
+    it('rejected (conn)', function()
+      local config = {
+        limiters = {
+          {name = "connections", key = 'test1', conn = 1, burst = 0, delay = 0.5}
+        },
+        redis_url = 'redis://localhost:6379/1'
+      }
+      local rate_limit_policy = RateLimitPolicy.new(config)
+      rate_limit_policy:access()
+      rate_limit_policy:access()
+      assert.spy(ngx_exit_spy).was_called_with(429)
+    end)
+    it('rejected (req)', function()
+      local config = {
+        limiters = {
+          {name = "leaky_bucket", key = 'test1', rate = 1, burst = 0}
+        },
+        redis_url = 'redis://localhost:6379/1'
+      }
+      local rate_limit_policy = RateLimitPolicy.new(config)
+      rate_limit_policy:access()
+      rate_limit_policy:access()
+      assert.spy(ngx_exit_spy).was_called_with(429)
+    end)
+    it('rejected (count)', function()
+      local config = {
+        limiters = {
+          {name = "fixed_window", key = 'test1', count = 1, window = 10}
+        },
+        redis_url = 'redis://localhost:6379/1'
+      }
+      local rate_limit_policy = RateLimitPolicy.new(config)
+      rate_limit_policy:access()
+      rate_limit_policy:access()
+      assert.spy(ngx_exit_spy).was_called_with(429)
+    end)
+    it('delay (conn)', function()
+      local config = {
+        limiters = {
+          {name = "connections", key = 'test1', conn = 1, burst = 1, delay = 2}
+        },
+        redis_url = 'redis://localhost:6379/1'
+      }
+      local rate_limit_policy = RateLimitPolicy.new(config)
+      rate_limit_policy:access()
+      rate_limit_policy:access()
+      assert.spy(ngx_sleep_spy).was_called_more_than(0.001)
```

Comment:
This is not correct. `was_called_more_than` refers to the number of times the spy was called. However, what you need is to check that it was called with a value that is > 0.001.

Same in the test below this one.